src/memex/storage.py

```diff
@@ -108,14 +109,16 @@ def create_annotation(request, data):
             )
 
     # The user must have permission to create an annotation in the group
-    # they've asked to create one in.
-    if data['groupid'] != '__world__':
-        group_principal = 'group:{}'.format(data['groupid'])
-        if group_principal not in request.effective_principals:
+    # they've asked to create one in. If the application didn't configure
+    # a groupfinder we will allow writing this annotation without any
+    # further checks.
+    groupfinder = request.registry.get(GROUPFINDER_KEY)
+    if groupfinder is not None:
```

Comment:
Note also that the way pyramid's configuration system works, you can actually *call* `memex_set_groupfinder` in `memex/groups.py`'s includeme in order to set the default. Because `h` includes this module (indirectly) it will then be able to override that call with its own call to `memex_set_groupfinder`.